lib/plugins/autolinks.js

```diff
@@ -13,23 +13,33 @@ module.exports = class Autolinks extends Diffable {
   }
 
   comparator (existing, attr) {
-    return existing.key_prefix === attr.key_prefix && existing.url_template === attr.url_template
+    return existing.key_prefix === attr.key_prefix &&
+      existing.url_template === attr.url_template
   }
 
   changed (existing, attr) {
-    return existing.key_prefix === attr.key_prefix && existing.url_template !== attr.url_template
+    // is_alphanumeric was added mid-2023. In order to continue to support settings yamls which dont specify this
+    // attribute, consider an unset is_alphanumeric as `true` (since that is the default value in the API)
+    // https://docs.github.com/en/rest/repos/autolinks?apiVersion=2022-11-28#create-an-autolink-reference-for-a-repository
+    const isAlphaNumericMatch = attr.is_alphanumeric === undefined
+      ? existing.is_alphanumeric // === true, the default
+      : attr.is_alphanumeric === existing.is_alphanumeric
+    return existing.key_prefix === attr.key_prefix &&
+      existing.url_template !== attr.url_template &&
+      !isAlphaNumericMatch
```

Comment:
I think this condition is incorrect, should be
```suggestion
    return existing.url_template !== attr.url_template || !isAlphaNumericMatch
```

The `key_prefix` check in not required since this was already matched by the `comparator`?

Reply:
Hi @markjm @stevoland should this stay open? Or should we implement the change above and merge it?

Reply:
Hi @decyjphr, we've been using the above with the changes suggested (with unit tests) in our fork for 6 months with no issues. I'd be happy to raise a PR next week.

We have a few more additions/fixes that we'd like to push upstream but would be more keen to contribute if the unit test at least were checked on CI.